src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/external/SentenceCalculationType.kt

```diff
@@ -1,150 +1,187 @@
-  SEC275(isSupported = false, isIndeterminate = true),
-  ALP_CODE21(isSupported = false, isIndeterminate = true),
-  LR_DPP(isSupported = false, isIndeterminate = true),
-  LR_ALP_CDE21(isSupported = false, isIndeterminate = true),
-  LR_ALP_LASPO(isSupported = false, isIndeterminate = true),
-  ZMD(isSupported = false, isIndeterminate = true),
-  SEC93(isSupported = false, isIndeterminate = true),
-  TWENTY(isSupported = false, isIndeterminate = true, primaryName = "20"),
-  NP(isSupported = false, isIndeterminate = false),
-  LR_EPP(isSupported = false, isIndeterminate = false),
-  CR(isSupported = false, isIndeterminate = false),
-  BOTUS(isSupported = true, isIndeterminate = false, sentenceClazz = BotusSentence::class.java),
-  AR(isSupported = false, isIndeterminate = false),
-  EPP(isSupported = false, isIndeterminate = false),
-  CUR_ORA(isSupported = false, isIndeterminate = false),
-  A_FINE(isSupported = false, isIndeterminate = false, primaryName = "A/FINE"),
-  CUR(isSupported = false, isIndeterminate = false),
-  CIVIL(isSupported = false, isIndeterminate = false),
-  EXT(isSupported = false, isIndeterminate = false),
-  LR_ES(isSupported = false, isIndeterminate = false),
-  YRO(isSupported = false, isIndeterminate = false),
-  SEC91(isSupported = false, isIndeterminate = false),
-  VOO(isSupported = false, isIndeterminate = false),
-  TISCS(isSupported = false, isIndeterminate = false),
-  STS21(isSupported = false, isIndeterminate = false),
-  STS18(isSupported = false, isIndeterminate = false),
-  UNIDENTIFIED(isSupported = false),
+  HDR_ORA(sentenceType = Unsupported, recallType = STANDARD_RECALL_255),
+  HDR(sentenceType = Unsupported, recallType = STANDARD_RECALL_255),
+  CUR(sentenceType = Unsupported, recallType = STANDARD_RECALL_255),
+  CUR_ORA(sentenceType = Unsupported, recallType = STANDARD_RECALL_255),
+  //endregion
+
+  //region Unidentified
+  UNIDENTIFIED(sentenceType = Unsupported),
+  //endregion
   ;
 
   companion object {
+
     fun from(sentenceCalculationType: String): SentenceCalculationType =
       entries.firstOrNull { it.primaryName == sentenceCalculationType }
         ?: entries.firstOrNull { it.name == sentenceCalculationType }
         ?: UNIDENTIFIED
 
     fun isSupported(sentenceCalculationType: String): Boolean =
+      !(isIndeterminate(sentenceCalculationType) || isUnsupportedSentence(sentenceCalculationType))
+
+    fun isIndeterminate(sentenceCalculationType: String): Boolean =
       try {
-        from(sentenceCalculationType).isSupported
+        from(sentenceCalculationType).sentenceType == Indeterminate
       } catch (error: IllegalArgumentException) {
         false
       }
 
-    fun isIndeterminate(sentenceCalculationType: String): Boolean =
+    private fun isUnsupportedSentence(sentenceCalculationType: String): Boolean =
       try {
-        from(sentenceCalculationType).isIndeterminate
+        from(sentenceCalculationType).sentenceType == Unsupported
```

Comment:
Is there any reason the isUnsupportedSentence() function doesn't return true for indeterminate sentences?